### PR TITLE
Account for null element

### PR DESCRIPTION
**Pull request by patsy-issa**
Adds null `target|element` detection and error. refs #736

### CODE REVIEW

File: lib/__tests__/does-not-have-tagname.ts [hunk 1]
```diff
@@ -59,14 +59,9 @@ describe('assert.dom(...).doesNotHaveTagName()', () => {
...
-
-      expect(assert.results).toEqual([
-        {
-          message: 'Element <unknown> should exist',
-          result: false,
-        },
-      ]);
+      expect(() => assert.dom(null).doesNotHaveTagName('div')).toThrow(
+        'Null target or element was passed'
+      );
```
Comment (Turbo87) [inline lib/__tests__/does-not-have-tagname.ts:64]: hmm... why change the existing support for `null` that we already have? 🤔

Comment (patsy-issa) [reply lib/__tests__/does-not-have-tagname.ts:64]: `assert.dom(null)` will be throwing the error before `doesNotHaveTagName` gets the chance ^^

File: lib/__tests__/does-not-have-tagname.ts [hunk 2]
```diff
@@ -100,17 +95,6 @@ describe('assert.dom(...).doesNotHaveTagName()', () => {
         },
       ]);
     });
-
-    test('fails for missing element', () => {
-      assert.dom('#missing').doesNotHaveTagName('div');
```
Comment (patsy-issa) [inline lib/__tests__/does-not-have-tagname.ts:105]: Ah I see these should remain, tunnel vision 🤦

File: lib/__tests__/does-not-have-tagname.ts (hunk 1 above)
Comment (patsy-issa) [reply lib/__tests__/does-not-have-tagname.ts:64]: @Turbo87 restored the valid missing element assertions

**Pull request closed without merging**